cmake/Extensions.cmake

```diff
@@ -22,10 +22,31 @@ define_property(GLOBAL PROPERTY EXTENSION-OPTIONS
 
 set_property(GLOBAL PROPERTY EXTENSION-OPTIONS "")
 
+set(extension-build-info-file "${CMAKE_CURRENT_BINARY_DIR}/ExtensionBuildInfo.cpp")
+file(GENERATE OUTPUT ${extension-build-info-file}
+    CONTENT "\
+    #include \"utils/Export.h\"\n\
+    #ifdef BUILD_ID_VARIABLE_NAME\n\
+    EXTENSIONAPI extern const char* const BUILD_ID_VARIABLE_NAME = \"__EXTENSION_BUILD_IDENTIFIER_BEGIN__${BUILD_IDENTIFIER}__EXTENSION_BUILD_IDENTIFIER_END__\";\n\
+    #else\n\
+    static_assert(false, \"BUILD_ID_VARIABLE_NAME is not defined\");\n\
+    #endif\n")
```

Comment:
I think this is a hack. We shouldn't have to open the extension as a binary file, search for a prefix and extract the ID as text. Why don't extensions define a function that takes a build identifier and returns whether the extension is compatible with the agent or not? The agent could call this function after dlopen, before the extension's init.

Reply:
by the time we would call the verification function in the dll, the damage could very well be done, as there are numerous static objects (loggers, processors) created at load-time registering themselves, calling into libminifi